libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Skills/AdaptiveSkillDialog.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using AdaptiveExpressions.Properties;
+using Microsoft.Bot.Builder.Skills;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive.Skills
+{
+    public class AdaptiveSkillDialog : SkillDialog
+    {
+        [JsonProperty("$kind")]
+        public const string DeclarativeType = "Microsoft.AdaptiveSkillDialog";
+
+        [JsonConstructor]
+        public AdaptiveSkillDialog([CallerFilePath] string callerPath = "", [CallerLineNumber] int callerLine = 0)
+            : base(new SkillDialogOptions())
+        {
+            DialogOptions.Skill = new BotFrameworkSkill();
+            RegisterSourceLocation(callerPath, callerLine);
+        }
+
+        /// <summary>
+        /// Gets or sets an optional expression which if is true will disable this action.
+        /// </summary>
+        /// <example>
+        /// "user.age > 18".
+        /// </example>
+        /// <value>
+        /// A boolean expression. 
+        /// </value>
+        [JsonProperty("disabled")]
+        public BoolExpression Disabled { get; set; }
+
+        [JsonProperty("botId")]
+        public StringExpression BotId { get; set; } = "=settings.MicrosoftAppId";
+
+        [JsonProperty("skillHostEndpoint")]
+        public StringExpression SkillHostEndpoint { get; set; } = "=settings.SkillHostEndpoint";
+
+        [JsonProperty("skillAppId")]
+        public StringExpression SkillAppId { get; set; }
```

Comment:
we should consider defaulting this to "=settings.SkillAppId", also in the component schema.

Reply:
This will cause an issue, if user added multiple skills in one bot, there would be =settings.SkillAppId1 =settings.SkillAppId2...

Reply:
Yes. Same issue as multiple Luis apps that we already have a solution for via the LUIS build library and Cli. I provided that guidance via email.

Reply:
@vishwacsena is it reasonable to assume most bots will have a lot more skills configured and in use than luis apps? i think so. maybe we shouldn't see lubuild as the convention. 

what if we had accessors by name instead? `=settings.SkillName.skillAppId`

Reply:
That's a good idea (it does work with multiple skills) we can address the default in the a later PR